uninstall_zabbix.rb

```diff
@@ -0,0 +1,274 @@
+  # This is really gross because Ruby 1.9 lacks Array#to_h.
+  File.readlines(File.join(repo_dir, 'cluster.txt'))
+    .map { |line| Hash[*fields.zip(line.split(' ')).flatten(1)] }
+end
+
+def repo_dir
+  File.dirname(__FILE__)
+end
+
+def check_status(c, on_fail_msg, on_success_msg, dofail = false)
+  if !c.status.success?
+    fail_msg = on_fail_msg + ' ' + c.stdout + '\n' + c.stderr
+    if dofail
+      fail fail_msg
+    else
+      puts fail_msg
+    end
+  else
+    puts on_success_msg
+  end
+end
+
+def stop_zabbix_agent(chef_env, host)
+  c = Mixlib::ShellOut.new('./nodessh.sh', chef_env, host['ip_address'],
+                           'service zabbix-agent stop', 'sudo')
+  c.run_command
+  check_status(c, 'Could not stop zabbix agent on ' + host['ip_address'],
+               'Stopped zabbix agent on ' + host['ip_address'])
+end
+
+def stop_zabbix_server(chef_env, host)
+  c = Mixlib::ShellOut.new('./nodessh.sh', chef_env, host['ip_address'],
+                           'service zabbix-server stop', 'sudo')
+  c.run_command
+  check_status(c, 'Could not stop zabbix server on ' + host['ip_address'],
+               'Stopped zabbix server on ' + host['ip_address'])
+  confirm_service_down(chef_env, host, 'zabbix-server')
+end
+
+def stop_zabbix_agent_and_server(chef_env)
+  head_nodes.each do |host|
+    puts host['ip_address']
+    stop_zabbix_agent(chef_env, host)
+    stop_zabbix_server(chef_env, host)
+  end
+end
+
+def uninstall_zabbix_api_gem(chef_env)
+  head_nodes.each do |host|
+    puts host['ip_address']
+    c = Mixlib::ShellOut.new('./nodessh.sh', chef_env, host['ip_address'],
+                             '/opt/chef/embedded/bin/gem uninstall zabbixapi',
+                             'sudo')
+    c.run_command
+    check_status(c,
+                 'Could not uninstall zabbix api gem on' + host['ip_address'],
+                 'Uninstalled zabbix api gem on ' + host['ip_address'])
+  end
+end
+
+# rubocop:disable all
+def find_chef_env
+  o = Ohai::System.new
+  o.all_plugins
+
+  env_command =
+    Mixlib::ShellOut.new('sudo', 'knife',
+                         'node', 'show',
+                         o[:fqdn] || o[:ip_address], '-E',
+                         '-f', 'json')
+
+  env_command.run_command
+
+  unless env_command.status.success?
+    fail 'Could not retrieve Chef environment!\n' +
+      env_command.stdout + '\n' +
+      env_command.stderr
+  end
+
+  JSON.parse(env_command.stdout)['chef_environment']
+end
+
+def drop_zabbix_database(chef_env, vm_entry, password)
+  unless zabbix_table?(chef_env, vm_entry, password)
+    return nil
+  end
+  mysqlcmd = 'mysql -uroot -p' + password + ' -e \'drop database zabbix;\''
+  c = Mixlib::ShellOut.new('./nodessh.sh', chef_env, vm_entry['ip_address'],
+                           mysqlcmd, 'sudo')
+  c.run_command
+  check_status(c, 'Could not drop zabbix database ', 'Dropped zabbix database',
+               true)
+end
+
+def uninstall_zabbix(chef_env)
```

Comment:
Maybe for sanity's sake make a mention that this is to workaround Chef-BACH #441